### DISCUSSION TITLE
Comparing a JSON variable to null

### DISCUSSION THREAD

**Discussion by nmattela** [elkowar/eww#1105] [Q&A]
I want to dynamically change the class of a button depending on a condition, in this case the condition is checking whether my variable `network-device` (a JSON string) is `null` (as in, the JSON null). I have the following code:
```
(button
    :class "${network-device == null ? "network-err" : "network-connected"}"
)
```
But eww doesn't quite understand what I mean by null:
```
error: No variable named `null` in scope
```

What would be the proper way to compare a JSON string to the JSON null?
Thanks in advance :)

**Comment by w-lfchen** [comment]
Try this:
```
(button
    :class {network-device == "null" ? "network-err" : "network-connected"}
)
```
Does this work?

**Reply by nmattela** [reply]
Yes, this does work, but then I am comparing with the string `"null"` as opposed to the JSON `null`, no?

**Reply by w-lfchen** [reply]
I have yet to exhaustively understand the internals of eww, so I can't give a definitive answer.
I think that internally everything is handled as strings and therefore, this is supported by my tests, when using JSON, the `null` value is the same as the JSON string `"null"`.
